tests/test_schema.py

```diff
@@ -197,3 +197,37 @@ def test_normalize_std(x):
 def test_normalize_error():
     with pytest.raises(SchemaError):
         assert normalize_field_type('thingy')
+
+
+@requires_only_gdal1
+def test_check_schema_driver_support(tmpdir):
+
+    # Shapefile
+    for field_type in ['datetime', 'time']:
+        try:
+            name = str(tmpdir.join('test_scheme.shp'))
+            items = [('field1',  field_type)]
+            with fiona.open(name, 'w',
+                            driver="ESRI Shapefile",
+                            schema={
+                                'geometry': 'LineString',
+                                'properties': items}) as c:
+                assert False
+            
+        except DriverSupportError:
+            assert True
+    
+    # GPKG
+    try:
+        name = str(tmpdir.join('test_scheme.gpkg'))
+        items = [('field1',  'time')]
+        with fiona.open(name, 'w',
+                        driver="GPKG",
+                        schema={
+                            'geometry': 'LineString',
+                            'properties': items}) as c:
+            assert False
+        
+    except DriverSupportError:
+        assert True
```

Comment:
Similarly, I'd like to see the geopackage case become its own proper test separate from the shapefile test.

Reply:
Refactored to a seperate test case